changelog/@unreleased/pr-5721.v2.yml

```diff
@@ -0,0 +1,5 @@
+type: improvement
+improvement:
+  description: Replace usages of testCompile for external dependencies.
```

Comment:
do we really need a changelog here?

Reply:
well it is okay to keep record of what changed - since we already have it do not want to remove it unless you feel strongly about this,

Reply:
This can be useful though I'd add a bit more context (specifically: users who encounter compile breaks here have likely transitively relied on AtlasDB pulling in dependencies - please declare them yourself).

Also, I don't think this matches what the PR does?

Reply:
(race-conditioned on second point)

Reply:
[I have no strong feelings one way or the other](https://www.youtube.com/watch?v=CxK_nA2iVXw) (hence the approval)